pkg/storage/stores/tsdb/head_manager.go

```diff
@@ -379,10 +378,11 @@ func (m *HeadManager) buildTSDBFromHead(head *tenantHeads) error {
 }
 
 func (m *HeadManager) truncateWALForPeriod(period int) (err error) {
-	m.metrics.tsdbWALTruncationsTotal.Inc()
 	defer func() {
-		if err != nil {
-			m.metrics.tsdbWALTruncationsFailedTotal.Inc()
+		if err == nil {
+			m.metrics.walTruncations.WithLabelValues(statusSuccess).Inc()
+		} else {
+			m.metrics.walTruncations.WithLabelValues(statusFailure).Inc()
 		}
```

Comment:
I tend to do it like this for better readability and reduce code repetition.
```suggestion
		status := statusSuccess
		if err == nil {
			status = statusFailure
		}
		m.metrics.walTruncations.WithLabelValues(status).Inc()
```